components/micro-gateway-core/src/main/ballerina/src/gateway/filters/subscription_filter.bal

```diff
@@ -43,137 +44,155 @@ public type SubscriptionFilter object {
 
 function doSubscriptionFilterRequest(http:Caller caller, http:Request request, http:FilterContext filterContext)
              returns boolean {
-    string authScheme = runtime:getInvocationContext().authContext.scheme;
-    printDebug(KEY_SUBSCRIPTION_FILTER, "Auth scheme: " + authScheme);
-    if (authScheme != AUTH_SCHEME_JWT){
-        printDebug(KEY_SUBSCRIPTION_FILTER, "Skipping since auth scheme != jwt.");
-        return true;
-    }
-    string jwtToken = runtime:getInvocationContext().authContext.authToken;
-    string currentAPIContext = getContext(filterContext);
-    AuthenticationContext authenticationContext = {};
-
-    json|error decodedPayload = {};
-    var cachedJwt = trap <auth:CachedJwt>jwtCache.get(jwtToken);
-    if (cachedJwt is auth:CachedJwt) {
-        printDebug(KEY_SUBSCRIPTION_FILTER, "jwt found from the jwt cache");
-        internal:JwtPayload jwtPayload = cachedJwt.jwtPayload;
-        json payload = {};
-        map<json> customClaims = jwtPayload.customClaims;
-        if(customClaims.hasKey(APPLICATION)) {
-            payload.application = customClaims[APPLICATION];
-        }
-        if(customClaims.hasKey(SUBSCRIBED_APIS)) {
-            payload.subscribedAPIs = customClaims[SUBSCRIBED_APIS];
-        }
-        if(customClaims.hasKey(CONSUMER_KEY)) {
-            payload.consumerKey = customClaims[CONSUMER_KEY];
-        }
-        if(customClaims.hasKey(KEY_TYPE)) {
-            payload.keytype = customClaims[KEY_TYPE];
-        }
-
-        payload.sub = jwtPayload["sub"];
-        decodedPayload = payload;
-    } else {
-        //If not found in cache decode jwt token and get the payload
-        var jwtPayload = getEncodedJWTPayload(jwtToken);
-        if (jwtPayload is error) {
-            log:printError(jwtPayload.reason(), err = jwtPayload);
-            setErrorMessageToFilterContext(filterContext, API_AUTH_GENERAL_ERROR);
-            sendErrorResponse(caller, request, filterContext);
-            return false;
-        } else {
-            printTrace(KEY_SUBSCRIPTION_FILTER, "Encoded JWT payload: " + jwtPayload);
-            decodedPayload = getDecodedJWTPayload(jwtPayload);
+    runtime:AuthenticationContext? authContext= runtime:getInvocationContext()?.authenticationContext;
+    if(authContext is runtime:AuthenticationContext) {
+        string authScheme = authContext.scheme;
+        printDebug(KEY_SUBSCRIPTION_FILTER, "Auth scheme: " + authScheme);
+        if (authScheme != AUTH_SCHEME_JWT){
+            printDebug(KEY_SUBSCRIPTION_FILTER, "Skipping since auth scheme != jwt.");
+            return true;
         }
-    }
+    
+        string jwtToken = authContext.authToken;
```

Comment:
`authContext` can be `()` here since we've skipped specifically checking `()` case.

Reply:
We are checking if authContext is of type AuthenticationContext. So here it can't be ()